homeassistant/components/intent.py

```diff
@@ -0,0 +1,178 @@
+"""Module to coordinate user intentions."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+
+DOMAIN = 'intent'
+_LOGGER = logging.getLogger(__name__)
+
+SLOT_SCHEMA = vol.Schema({
+    vol.Optional('type'): str,
+})
+
+SPEECH_TYPE_PLAIN = 'plain'
+SPEECH_TYPE_SSML = 'ssml'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup intent."""
+    hass.intent = IntentRegistry(hass)
```

Comment:
But I don't like this, while we go back like the old group have work before.
If we do that, we should use our get_component() function. If we have dependency set to "intent" component we already load it before we use `intent.register`, so we need not care about that.

I think we should use our dispatcher logic and maybe it make sense to use it for http too in future. We use this also for mqtt that is also a component like ident/http.

A other way, but I think that should be only for external library, is to make helper function around `hass.data` like aiohttp client.